packages/sanity/src/core/form/inputs/files/FileInput/FileInput.tsx

```diff
@@ -352,12 +345,14 @@ export class BaseFileInput extends React.PureComponent<BaseFileInputProps, BaseF
           isChanged={changed}
         >
           {/* not uploading */}
-          {!value?._upload && (
+          {value?._upload ? (
+            this.renderUploadState(value._upload)
+          ) : (
             <FileTarget
               {...elementProps}
+              onFocus={this.handleFileTargetFocus}
```

Comment:
We need this special handling in order to prevent interactive elements _inside_ the file target from triggering focus on the file input itself (see implementation of `this.handleFileTargetFocus`)